### ISSUE TITLE
If register the UpdateProgress event of ShellFileOperations, the operation will be invalid

### ISSUE THREAD

**Issue by zhuxb711** [dahall/Vanara#146]
**Describe the bug**
If there is no following line, the operation is performed normally
```csharp
 Operation.UpdateProgress += Progress;
```

If I register for the UpdateProgress event, the copy operation will be invalid. It means that no files are copied.

By the way, I want to know whether it is correct way to get the copy progress?

**What code is involved**
```csharp
        public static bool Copy(IEnumerable<string> Source, string DestinationPath, ProgressChangedEventHandler Progress)
        {
            try
            {
                if (!Directory.Exists(DestinationPath))
                {
                    _ = Directory.CreateDirectory(DestinationPath);
                }

                using (ShellFileOperations Operation = new ShellFileOperations
                {
                    Options = ShellFileOperations.OperationFlags.AddUndoRecord | ShellFileOperations.OperationFlags.NoConfirmMkDir | ShellFileOperations.OperationFlags.Silent
                })
                {
                    Operation.UpdateProgress += Progress;

                    foreach (string SourcePath in Source)
                    {
                        using (ShellItem SourceItem = new ShellItem(SourcePath))
                        using (ShellFolder DestItem = new ShellFolder(DestinationPath))
                        {
                            Operation.QueueCopyOperation(SourceItem, DestItem);
                        }
                    }

                    Operation.PerformOperations();
                }

                return true;
            }
            catch
            {
                return false;
            }
        }
```
**Expected behavior**
Copy files and report progress

**Comment by dahall** [comment]
This is the correct way. There was a bug which is now fixed that threw an exception thus causing the operation to not complete.

**Comment by zhuxb711** [comment]
When will a new version with this fix be released? Looking forward to your release.

**Comment by dahall** [comment]
In the next 2-3 days.

**Comment by dahall** [comment]
I have just released v3.2.12 with this fix

**Comment by zhuxb711** [comment]
Got it, now it work well, Thanks.
